sse_client.js

```diff
@@ -18,130 +18,132 @@ function isEmpty(obj) {
-        
+
+        node.headers = config.headers || {};
+        node.url = config.url;
+        node.events = config.events || [];
+        node.proxy = config.proxy;
+        node.restart = config.restart;
+        node.timeout = config.timeout;
+        node.rejectUnauthorized = config.rejectUnauthorized;
+        node.withCredentials = config.withCredentials;
+        node.paused = false;
+        node.prevMsg = null;
+        node.timerId = null;
+        node.client = null;
+        node.proxyUrl = null;
+
         // Migration of old nodes without the http settings
         if (node.rejectUnauthorized === undefined) {
             node.rejectUnauthorized = true;
         }
         if (node.withCredentials === undefined) {
             node.withCredentials = true;
         }
-        
-        var isTemplatedUrl = (node.url||"").indexOf("{{") != -1;
 
-        node.status({fill: 'red', shape: 'ring', text: 'disconnected'});
-        
+        var isTemplatedUrl = (node.url || "").indexOf("{{") != -1;
+
+        node.status({ fill: 'red', shape: 'ring', text: 'Disconnected. Wait for input msg to start.' });
+
         function handleEvent(e) {
             // Skip all events when this node is paused
             if (node.paused) {
                 return;
             }
-            
+
             // Skip the 'open' event
             if (e.type === 'open') {
                 return;
             }
-            
+
             // When events have been specified, only allow those events
             if (node.events.length > 0 && !node.events.includes(e.type)) {
                 return;
             }
-                    
-            // When a previous timer is available, stop it
-            if (node.timerId) {
-                clearTimeout(node.timerId);
-                node.timerId = null;
-            }
-            
+
             // When a timeout is specified, start a new timer (that restarts the SSE client)
-            if (node.restart) {
-                node.timerId = setTimeout(function restartClient() {
-                    // Restart the SSE client by resending the last message again
-                    handleMsg(node.prevMsg);
-                }, node.timeout * 1000);
-            }
-            
+            if (node.restart) startTimeoutTimer();
+
             // Send the received SSE event in the output message
             node.send({
                 event: e.type,
                 payload: e.data
             });
         }
-        
+
+        // 08/04/2022 Supergiovane: Function to that starts the timeout timer.
+        function startTimeoutTimer() {
+            // When a previous timer is available, stop it
+            if (node.timerId !== null) clearTimeout(node.timerId);
+            node.timerId = setTimeout(function restartClient() {
+                // Restart the SSE client by resending the last message again
+                node.status({ fill: "yellow", shape: "dot", text: "Timeout. Restarting." });
```

Comment:
Can you please change the status text to "timeout => restarting".  Read somewhere some time ago that status is lowercase, so I try to follow that (unofficial?) standard...

Reply:
Hi Bart
Sorry for that! I never read about lowercase status texts!
Oh my! All my nodes doesn't follow that rule!
Thank you for pointing me out! I'll slowly fix all my nodes with lowecase statuses.